integrations/oci/metrics/metrics/src/test/java/io/helidon/integrations/oci/metrics/OciMetricsSupportTest.java

```diff
@@ -262,45 +270,24 @@ private void validateMetricCount(String[] scopes, int expectedMetricCount) {
 
     private void validateMetricCount(OciMetricsSupport.Builder ociMetricsSupportBuilder, int expectedMetricCount) {
         testMetricCount = 0;
+        countDownLatch1 = new CountDownLatch(1);
         Routing routing = createRouting(ociMetricsSupportBuilder);
         WebServer webServer = createWebServer(routing);
 
-        Timer timer = new Timer(10);
-        while (testMetricCount <= 0) {
-            if (timer.expired()) {
-                fail(String.format("Timed out after %d sec. waiting for testMetricCount",
-                        timer.getTimeout()));
-            }
-            delay(50L);
+        try {
+            // Wait for signal from metric update that testMetricCount has been retrieved
+            countDownLatch1.await(10, TimeUnit.SECONDS);
+        } catch(InterruptedException e) {
+            fail("Error while waiting for testMetricCount: " + e.getMessage());
         }
-
+        webServer.shutdown().await(10, java.util.concurrent.TimeUnit.SECONDS);
         assertThat(testMetricCount, is(equalTo(expectedMetricCount)));
-        webServer.shutdown();
-    }
-
-    private static class Timer {
-        private final long endTime;
-        private final int timeOut;
-
-        public Timer(int timeOut) {
-            this.timeOut = timeOut;
-            this.endTime = System.nanoTime() + 1_000_000_000L * timeOut;
-        }
-
-        public boolean expired() {
-            return System.nanoTime() >= this.endTime;
-        }
-
-        int getTimeout() {
-            return this.timeOut;
-        }
     }
 
     private static void delay(long millis) {
         try {
             Thread.sleep(millis);
         } catch (InterruptedException ignore) {
-            Thread.currentThread().interrupt();
```

Comment:
Should this method at least do _something_ if it is interrupted? Maybe invoke `fail` to fail whatever test is in progress?

Reply:
Adding fail rather than ignoring.